lib/datasampler.rb

```diff
@@ -25,70 +41,116 @@ def delete_data
         Iam.settings.DB[table.to_sym].delete
       end
     end
-
     Iam.settings.DB[:projects].delete
     Iam.settings.DB[:clients].delete
   end
 
+  # Find the latest date in the information to import
+  # rubocop:disable MethodLength
+  def date_offset
+    latest_date = nil
+    Dir.foreach(@directory) do |file|
+      iterate_lines(@directory + file) do |line|
+        JSON.parse(line).each do |_key, value|
+          next unless value.is_a? String
+          begin
+            date = DateTime.parse value
+            latest_date = date if latest_date.nil? || date > latest_date
+          rescue ArgumentError
+            next
+          end
+        end
+      end
+    end
+    @date_offset = DateTime.now - latest_date
+  end
+
   # Imports the clients from clients.json
   def import_clients
-    # get clients from file, import to Client model
     clients_filename = @directory + 'clients.json'
-    clients_json = File.open(clients_filename, &:readline)
-    clients = Client.array_from_json(clients_json,
-                                     fields: Client.columns.map(&:to_s))
-
-    clients.each(&:save)
+    Client.unrestrict_primary_key
+    iterate_lines(clients_filename) do |line|
```

Comment:
I'm doing it line by line in an effort to speed it up a little. Beforehand, we were reading the entire file into memory first, then iterating over it. This way, much less RAM is used. I optimized this further in a later commit by aggregating the result of the change in an array, then using Sequel's import function to reduce the number of transactions used